scripts/metadataProvider/createPseudoModulesInfo.ts

```diff
@@ -0,0 +1,138 @@
+import {pipeline} from "node:stream/promises";
+import {Extract} from "unzip-stream";
+import MetadataProvider from "./MetadataProvider.js";
+import {writeFile} from "node:fs/promises";
+
+import type {UI5Enum, UI5EnumValue} from "@ui5-language-assistant/semantic-model-types";
+
+const RAW_API_JSON_FILES_FOLDER = "tmp/apiJson";
+
+async function downloadAPIJsons(url: string) {
+	const response = await fetch(url);
+	if (!response.ok) {
+		throw new Error(`unexpected response ${response.statusText}`);
+	}
+
+	if (response.body && response.body instanceof ReadableStream) {
+		await pipeline(response.body, Extract({path: RAW_API_JSON_FILES_FOLDER}));
+	} else {
+		throw new Error("Malformed response");
+	}
+}
+
+async function transformFiles() {
+	const metadataProvider = new MetadataProvider();
+	await metadataProvider.init(RAW_API_JSON_FILES_FOLDER);
+
+	const {enums} = metadataProvider.getModel();
+
+	const groupedEnums = Object.keys(enums).reduce((acc: Record<string, UI5Enum[]>, enumKey: string) => {
+		const curEnum = enums[enumKey];
+
+		acc[curEnum.library] = acc[curEnum.library] ?? [];
+		acc[curEnum.library].push(curEnum);
+
+		return acc;
+	}, Object.create(null) as Record<string, UI5Enum[]>);
+
+	await addOverrides(groupedEnums);
+}
+
+function buildJSDoc(enumEntry: UI5Enum | UI5EnumValue, indent = "") {
+	const jsDocBuilder: string[] = [`${indent}/**`];
+
+	if (enumEntry.description) {
+		jsDocBuilder.push(`${indent} * ${enumEntry.description.replaceAll("\n", "\n" + indent + " * ")}`);
+		jsDocBuilder.push(`${indent} *`);
+	}
+
+	if (enumEntry.experimentalInfo) {
+		let experimental = `${indent} * @experimental`;
+		if (enumEntry.experimentalInfo.since) {
+			experimental += ` (since ${enumEntry.experimentalInfo.since})`;
+		}
+		if (enumEntry.experimentalInfo.text) {
+			experimental += ` - ${enumEntry.experimentalInfo.text}`;
+		}
+		jsDocBuilder.push(experimental);
+	}
+
+	if (enumEntry.deprecatedInfo) {
+		let deprecated = `${indent} * @deprecated`;
+		if (enumEntry.deprecatedInfo.since) {
+			deprecated += ` (since ${enumEntry.deprecatedInfo.since})`;
+		}
+		if (enumEntry.deprecatedInfo.text) {
+			deprecated += ` - ${enumEntry.deprecatedInfo.text}`;
+		}
+		jsDocBuilder.push(deprecated);
+	}
+
+	if (enumEntry.visibility) {
+		jsDocBuilder.push(`${indent} * @${enumEntry.visibility}`);
+	}
+
+	if (enumEntry.since) {
+		jsDocBuilder.push(`${indent} * @since ${enumEntry.since}`);
+	}
+	jsDocBuilder.push(`${indent}*/`);
+
+	return jsDocBuilder.join("\n");
+}
+
+async function addOverrides(enums: Record<string, UI5Enum[]>) {
+	const indexFilesImports: string[] = [];
+
+	for (const libName of Object.keys(enums)) {
+		const enumEntries = enums[libName];
+		const stringBuilder: string[] = [];
+
+		enumEntries.forEach((enumEntry) => {
```

Comment:
Currently, this also generates declarations for enums that are actually modules on their own (e.g. "sap/ui/core/AnimationMode", which exists as module and within sap/ui/core/library.js; or "sap/ui/core/BindingMode", which only exists on its own and is not a pseudo-module).

Reply:
Indeed, I was wondering how to filter them and the only solution so far was to check the `resource` property in the raw API.json files